### DISCUSSION TITLE
A light theme with pure white background and higher contrast

### DISCUSSION THREAD

**Discussion by sriranggd** [zed-industries/zed#6485] [Feature Requests]
Hi,

Fortunately or unfortunately, my primary sitting area has a lot of sunlight falling on the screen because of which dark themes don't work for me. I have been using the light theme on VS-Code and Idea IntelliJ for quite some time now. Both of those have a white theme with a pure white background and pretty good contrast for the text colours.

The closest that I see in Zed is "One Light" but (as several others have mentioned here in the discussions) it looks a bit washed out (probably because of low contrast and not having a pure white background).

Anyway for me to get a light theme like the one provided by VS-Code?

**Comment by SilentGlasses** [comment]
Pretty sure if you add this config to your `~/.config/zed/settings.json` file you will find one you like....

```
"experiments": {
    "experimental_themes": true
}
```

Here's what mine looks like:

```
// ~/.config/zed/settings.json
{
    "theme": "One Dark",
    "buffer_font_size": 16,
    "experiments": {
        "experimental_themes": true
    }
}
```

**Comment by JosephTLyons** [comment]
Hey @sriranggd, curious, which VS Code theme specifically do you prefer?  At some point, Zed will have the ability to accept user-defined themes.

**Reply by petros** [reply]
I wonder if a high contrast light one would help in this scenario, @sriranggd.

**Reply by sriranggd** [reply]
Apologies for the delayed response. I missed the notifications.. ! What works best for me on VS-Code is the `Light+ (default light)` theme. If I can use it, nothing like it..!

<img width="602" alt="Screenshot 2023-03-15 at 6 20 09 PM" src="https://user-images.githubusercontent.com/310354/225313831-a3dd99c2-7046-4ff4-87cd-8b4f618102ab.png">

The closest one that I have seen on Zed so far is `One Light` but that one is also very low contrast. It feels a bit blurry and not vivid.

**Comment by iamnbutler** [comment]
Hey! I work on Zed's design – We are aware of contrast issues. Behind the scenes we are pushing on a multi-part system that should banish this problem forever, but it will take some time. Uninsured estimate: Sometime around June. 

In the meantime, I may ship a couple of new themes when we land overrides in the next few weeks. Would the [Github Light](https://user-images.githubusercontent.com/378023/132220037-3cd3e777-55a6-445f-9a2e-da6020ebd78d.png) theme fulfill your needs if we shipped it soon-ish?

**Reply by photex** [reply]
In Vs Code I use "GitHub Light Monochrome Focused". In Xcode I use "Low Key" and this is my absolute favorite. Extremely comfortable in multiple lighting conditions. I "ported" that to CLion even (but haven't gotten around to VS Code yet).

**Reply by photex** [reply]
Here are examples of my themes in case it's useful reference for anyone.

Xcode:

<img width="889" alt="Screenshot 2023-03-12 at 09 58 19" src="https://user-images.githubusercontent.com/301903/224534978-f4e643cc-dd6d-48f7-a1e6-fed6cfc348bd.png">

VS Code:

<img width="945" alt="Screenshot 2023-03-12 at 09 59 26" src="https://user-images.githubusercontent.com/301903/224535013-baebbf75-c595-4845-8030-1caceeba802c.png">

CLion:

<img width="945" alt="Screenshot 2023-03-12 at 10 00 21" src="https://user-images.githubusercontent.com/301903/224535022-ef19d65e-2d70-4dec-afcc-419249ac71af.png">

**Reply by sriranggd** [reply]
@iamnbutler Thanks for the response and sorry for the delay in responding. From the screenshot image you have shared, it looks like Github Light might work. It appears very similar to the `Light+` theme on VS-Code that I use. For your reference here is how the two editors look on my system : 

<img width="828" alt="Screenshot 2023-03-15 at 6 17 05 PM" src="https://user-images.githubusercontent.com/310354/225315005-7b1d09bd-a87c-4235-a923-ce3464cb0ff0.png">
<img width="768" alt="Screenshot 2023-03-15 at 6 16 35 PM" src="https://user-images.githubusercontent.com/310354/225315024-ac40b091-065e-45ff-92e9-e97705c50d39.png">
<img width="1440" alt="Screenshot 2023-03-15 at 6 17 38 PM" src="https://user-images.githubusercontent.com/310354/225315043-42c103b5-721d-4ffc-b724-56e8df47e52c.png">

**Reply by sriranggd** [reply]
Please note that while the above screenshots show the code colours, the same goes to the rest of the editor UI also (like the file browser / tab bar / command pallette etc). If we can get some high contrast in those too, it would be great.. !

**Comment by iamnbutler** [comment]
Updated versions of one dark/light that should almost exactly match the canonical Atom versions should land in the next two weeks. 

Let me know if these meet your needs!

**Reply by sriranggd** [reply]
Great..!

Have these landed now? I guess not. I am on `0.77.3` and the screenshots I have shared above are from this version.

**Comment by JosephTLyons** [comment]
We now have a ton of theme extensions - maybe there is something that works for you now!

**Reply by sriranggd** [reply]
Been on macOS Classic Light. That comes close. The colours are not so appealing to me but that's more of a personalization thing. So this issue can be closed now.

**Comment by MadLittleMods** [comment]
I just went through this exact same user journey. All of the current light themes for Zed are too dull/dim/muted and washed out.

I ended up creating a port of the VSCode Light+ theme for Zed :rocket:: https://github.com/MadLittleMods/zed-theme-vscode-light-plus

Zed on Linux (dim) | Zed on macOS (normal) | VSCode on Linux (baseline)
--- | --- | ---
![Zed on Linux has dim and washed out colors](https://github.com/user-attachments/assets/9dddc73c-4ada-44f8-994a-c2a0e1b34054) | <img alt="Zed on macOS looks normal" src="https://github.com/user-attachments/assets/e1cdb95e-0bd0-4393-99c3-ca0e5ab8de09" width="1230"> | ![VSCode example with Light+](https://github.com/user-attachments/assets/3d1588a8-7ce7-4599-a23b-2f5545d877a3)
![Zed on Linux has dim and washed out colors](https://github.com/user-attachments/assets/7db9d1c6-dd93-4d42-a27d-a854802f5335) | <img alt="Zed on macOS looks normal" src="https://github.com/user-attachments/assets/9e934027-3d42-4e79-bebc-46cdbc0b4a89" width="1230"> | ![VSCode example with Light+](https://github.com/user-attachments/assets/7c56a5f2-5060-43c5-891a-e1bb3ae754ef)


The dim and washed out color issue seems to be a specific problem to Linux (Manjaro with XFCE, arch-based on a non-retina monitor) as things appear fine on macOS. The screenshots are using the [`Fira Code`](https://github.com/tonsky/FiraCode) font (via the `ttf-fira-code` package) but I was also seeing the same problem with whatever default Zed font on Linux. I've cross-linked a bunch more context in a [*current limitations* section](https://github.com/MadLittleMods/zed-theme-vscode-light-plus#dim-and-washed-out-colors) in the theme readme.

 - https://github.com/zed-industries/zed/issues/9057
 - https://github.com/zed-industries/zed/issues/19100
 - https://github.com/zed-industries/zed/discussions/6485#discussioncomment-4937330
 - https://github.com/zed-industries/zed/issues/17890
 - https://github.com/zed-industries/zed/issues/8293#issuecomment-2299761507
 - https://github.com/zed-industries/zed/issues/15521
 - Even the theme previews on [zed-themes.com](https://zed-themes.com/) are much more vibrant because the browser is rendering things as expected.
